--- protobuf-3.20.3/src/google/protobuf/port_def.inc	2022-09-29 21:09:51.000000000 +0000
+++ protobuf-3.20.3/src/google/protobuf/port_def.inc	2025-08-27 14:16:29.721253876 +0000
@@ -233,9 +233,8 @@
 #ifdef PROTOBUF_TAILCALL
 #error PROTOBUF_TAILCALL was previously defined
 #endif
-#if __has_cpp_attribute(clang::musttail) && !defined(__arm__) && \
-    !defined(_ARCH_PPC) && !defined(__wasm__) &&                 \
-    !(defined(_MSC_VER) && defined(_M_IX86))
+#if __has_cpp_attribute(clang::musttail) && (defined(__aarch64__) || \
+    defined(__x86_64__) || defined(_M_X64))
 #  ifndef PROTO2_OPENSOURCE
 // Compilation fails on ARM32: b/195943306
 // Compilation fails on powerpc64le: b/187985113
