--- protobuf-3.17.3/ruby/ext/google/protobuf_c/extconf.rb~	2020-02-14 21:13:37.000000000 +0100
+++ protobuf-3.17.3/ruby/ext/google/protobuf_c/extconf.rb	2020-04-07 22:56:24.419874091 +0200
@@ -3,19 +3,19 @@
 require 'mkmf'
 
 if RUBY_PLATFORM =~ /darwin/ || RUBY_PLATFORM =~ /linux/
-  $CFLAGS += " -std=gnu99 -O3 -DNDEBUG -fvisibility=hidden -Wall -Wsign-compare -Wno-declaration-after-statement"
+  $CFLAGS += " -std=gnu99 -DNDEBUG -fvisibility=hidden -Wall -Wsign-compare -Wno-declaration-after-statement"
 else
-  $CFLAGS += " -std=gnu99 -O3 -DNDEBUG"
+  $CFLAGS += " -std=gnu99 -DNDEBUG"
 end
 
 
 if RUBY_PLATFORM =~ /linux/
   # Instruct the linker to point memcpy calls at our __wrap_memcpy wrapper.
-  $LDFLAGS += " -Wl,-wrap,memcpy"
+  $LDFLAGS += " -std=c99 -DNDEBUG"
 end
 
 $objs = ["protobuf.o", "convert.o", "defs.o", "message.o",
-         "repeated_field.o", "map.o", "ruby-upb.o", "wrap_memcpy.o"]
+         "repeated_field.o", "map.o", "ruby-upb.o"]
 
 find_header('third_party/wyhash/wyhash.h', '../../../..')
 create_makefile("google/protobuf_c")
