--- protobuf-3.6.1/ruby/ext/google/protobuf_c/extconf.rb.orig	2018-10-12 18:23:59.206483460 +0200
+++ protobuf-3.6.1/ruby/ext/google/protobuf_c/extconf.rb	2018-10-12 18:42:35.729804045 +0200
@@ -3,17 +3,11 @@
 require 'mkmf'
 
 unless RUBY_PLATFORM =~ /mswin|mingw/
-  $CFLAGS += " -std=c99 -O3 -DNDEBUG"
+  $CFLAGS += " -std=c99 -DNDEBUG"
 end
 
 
-if RUBY_PLATFORM =~ /linux/
-  # Instruct the linker to point memcpy calls at our __wrap_memcpy wrapper.
-  $LDFLAGS += " -Wl,-wrap,memcpy"
-end
-
 $objs = ["protobuf.o", "defs.o", "storage.o", "message.o",
-         "repeated_field.o", "map.o", "encode_decode.o", "upb.o",
-         "wrap_memcpy.o"]
+         "repeated_field.o", "map.o", "encode_decode.o", "upb.o"]
 
 create_makefile("google/protobuf_c")
