--- protobuf-3.11.4/ruby/ext/google/protobuf_c/extconf.rb~	2020-02-14 21:13:37.000000000 +0100
+++ protobuf-3.11.4/ruby/ext/google/protobuf_c/extconf.rb	2020-04-07 22:56:24.419874091 +0200
@@ -3,19 +3,17 @@
 require 'mkmf'
 
 if RUBY_PLATFORM =~ /darwin/ || RUBY_PLATFORM =~ /linux/
-  $CFLAGS += " -std=gnu90 -O3 -DNDEBUG -Wall -Wdeclaration-after-statement -Wsign-compare"
+  $CFLAGS += " -std=gnu90 -DNDEBUG -Wall -Wdeclaration-after-statement -Wsign-compare"
 else
-  $CFLAGS += " -std=gnu90 -O3 -DNDEBUG"
+  $CFLAGS += " -std=gnu90 -DNDEBUG"
 end
 
 
 if RUBY_PLATFORM =~ /linux/
-  # Instruct the linker to point memcpy calls at our __wrap_memcpy wrapper.
-  $LDFLAGS += " -Wl,-wrap,memcpy"
+  $LDFLAGS += " -std=c99 -DNDEBUG"
 end
 
 $objs = ["protobuf.o", "defs.o", "storage.o", "message.o",
-         "repeated_field.o", "map.o", "encode_decode.o", "upb.o",
-         "wrap_memcpy.o"]
+         "repeated_field.o", "map.o", "encode_decode.o", "upb.o" ]
 
 create_makefile("google/protobuf_c")
